internal/infrastructure/logger/init.go

```diff
@@ -1,18 +1,94 @@
 package logger
 
 import (
+	"os"
+	"strings"
+
 	"github.com/podengo-project/idmsvc-backend/internal/config"
-	"github.com/rs/zerolog"
+	"golang.org/x/exp/slog"
+)
+
+const (
+	LevelTrace  = slog.Level(-8)
+	LevelDebug  = slog.LevelDebug
+	LevelInfo   = slog.LevelInfo
+	LevelNotice = slog.Level(2)
+	LevelWarn   = slog.LevelWarn
+	LevelError  = slog.LevelError
 )
 
+type AppHandler struct {
+	handler slog.Handler
+}
+
+type Clonable interface {
+	Clone() interface{}
+}
+
+// If you want to learn more about slog:
+// https://betterstack.com/community/guides/logging/logging-in-go/
 func InitLogger(cfg *config.Config) {
 	if cfg == nil {
 		panic("'cfg' cannot be nil")
 	}
 
-	lvl, err := zerolog.ParseLevel(cfg.Logging.Level)
-	if err != nil {
-		panic(err)
+	var h slog.Handler
+
+	globalLevel := new(slog.LevelVar)
+	// set default to warning
+	globalLevel.Set(LevelWarn)
+
+	LevelNames := map[slog.Leveler]string{
+		LevelTrace:  "TRACE",
+		LevelNotice: "NOTICE",
+	}
+
+	opts := slog.HandlerOptions{
+		AddSource: false,
+		Level:     globalLevel,
+		// This will print TRACE and NOTICE in logs nicely
+		ReplaceAttr: func(groups []string, a slog.Attr) slog.Attr {
+			if a.Key == slog.LevelKey {
+				level := a.Value.Any().(slog.Level)
+				levelLabel, exists := LevelNames[level]
+				if !exists {
+					levelLabel = level.String()
+				}
+
+				a.Value = slog.StringValue(levelLabel)
+			}
+
+			return a
+		},
+	}
+
+	if cfg.Logging.Console {
+		h = slog.NewTextHandler(
+			os.Stderr,
+			&opts,
+		)
+	} else {
+		h = slog.NewJSONHandler(
+			os.Stderr,
+			&opts,
+		)
+	}
+	slog.SetDefault(slog.New(h))
+
+	// set global log level
+	lvl := strings.ToUpper(cfg.Logging.Level)
+
+	if lvl == "TRACE" {
+		globalLevel.Set(LevelTrace)
+	} else if lvl == "DEBUG" {
+		globalLevel.Set(LevelDebug)
+	} else if lvl == "INFO" {
+		globalLevel.Set(LevelInfo)
+	} else if lvl == "NOTICE" {
+		globalLevel.Set(LevelNotice)
+	} else if lvl == "WARN" {
+		globalLevel.Set(LevelWarn)
+	} else if lvl == "ERROR" {
+		globalLevel.Set(LevelError)
```

Comment:
Contrary to C, you can use a switch statement for string pattern matching. Easier to read.

Reply:
Changed to swtich.

Reply:
the current state updated the chained if by a switch statement; resolving this conversation.